osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/DefaultEncryptionDelegate.java

```diff
@@ -28,16 +28,23 @@ public DefaultEncryptionDelegate(final List<EncryptionProvider> encryptionProvid
     }
 
     @Override
-    public EncryptedSecret encrypt(final EncryptionProviderType encryptionProviderType, final Secret secret,
-            final String keyReference) {
+    public EncryptedSecret encrypt(final EncryptionProviderType encryptionProviderType, final byte[] secret,
+            final String keyReference) throws EncrypterException {
         return this.providers.stream().filter(ep -> ep.getType().equals(encryptionProviderType)).findFirst().orElseThrow(
                 () -> new EncrypterException(ERROR_NO_PROVIDER)).encrypt(secret, keyReference);
     }
 
     @Override
-    public Secret decrypt(final EncryptedSecret secret, final String keyReference) {
+    public byte[] decrypt(final EncryptedSecret secret, final String keyReference) throws EncrypterException {
         return this.providers.stream().filter(ep -> ep.getType().equals(secret.getType())).findFirst().orElseThrow(
                 () -> new EncrypterException(ERROR_NO_PROVIDER)).decrypt(secret, keyReference);
     }
+
+    @Override
+    public byte[] generateAes128BitsSecret(final EncryptionProviderType encryptionProviderType,
+            final String keyReference) throws EncrypterException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.opensmartgridplatform.shared.exceptionhandling.EncrypterException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)